### ISSUE TITLE
`IndexOutOfBoundsException` when using Mockito's mocking/spying without specifying class

### ISSUE THREAD

**Issue by rdesgroppes** [google/error-prone#3609]
[Since Mockito 4.[10].0](https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html#54):
> Instead of calling method [`mock(Class)`](https://javadoc.io/static/org.mockito/mockito-core/4.10.0/org/mockito/Mockito.html#mock(java.lang.Class)) or [`spy(Class)`](https://javadoc.io/static/org.mockito/mockito-core/4.10.0/org/mockito/Mockito.html#spy(java.lang.Class)) with Class parameter, you can now now call method `mock()` or `spy()` **without parameters**:
> ```java
>    Foo foo = Mockito.mock();
>    Bar bar = Mockito.spy();
>  ```
> Mockito will automatically detect the needed class.

This feature isn't supported by Error Prone, since using it leads to:
```
/path/to/File.java: error: An unhandled exception was thrown by the Error Prone static analysis plugin.
        final Supplier<String> stringSupplier = mock();
                               ^
     Please report this at https://github.com/google/error-prone/issues/new and include the following:
  
     error-prone version: 2.11.0
     BugPattern: DoNotMock
     Stack Trace:
     java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
  	at jdk.compiler/com.sun.tools.javac.util.List.get(List.java:490)
  	at com.google.errorprone.bugpatterns.AbstractMockChecker.lambda$extractFirstArg$2(AbstractMockChecker.java:172)
```
... which imposes to disable `DoNotMock` either locally or globally for the time being.

**Comment by Stephan202** [comment]
There's a PR in the works to improve this, IIUC: #3602.

**Comment by graememorgan** [comment]
Fixed by #3602.
